### DISCUSSION TITLE
generate black images by FLUX

### DISCUSSION THREAD

**Discussion by movelikeriver** [comfyanonymous/ComfyUI#4348] [Q&A]
the default workflow (ComfyUI) can generate file successfully on my laptop.

followed this FLUX instruction https://youtu.be/Qiltbxvhr_A?feature=shared 
made this workflow [workflow_flux.json](https://github.com/user-attachments/files/16608370/workflow_flux.json)
it always generate pure black image.

OS: apple mac, Sonoma 14.4, M3 pro, 36G ram

didn't see obvious error info
```
got prompt
Using split attention in VAE
Using split attention in VAE
model weight dtype torch.bfloat16, manual cast: None
model_type FLOW
Requested to load FluxClipModel_
Loading 1 new model
Requested to load AutoencodingEngine
Loading 1 new model
Unloading models for lowram load.
0 models unloaded.
Requested to load Flux
Loading 1 new model
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [07:59<00:00, 23.95s/it]
Requested to load AutoencodingEngine
Loading 1 new model
Prompt executed in 626.45 seconds
```

init loading info:
```
Total VRAM 36864 MB, total RAM 36864 MB
pytorch version: 2.3.1
Set vram state to: SHARED
Device: mps
Using sub quadratic optimization for cross attention, if you have memory or speed issues try using: --use-split-cross-attention
[Prompt Server] web root: /Users/ratc/mycode/ml/diffusion/ComfyUI/web
### Loading: ComfyUI-Manager (V2.48.8)
### ComfyUI Revision: 2534 [3e52e036] | Released on '2024-08-13'
### Loading: ComfyUI-Impact-Pack (V6.2)
```

thanks in advance!

**Comment by BrechtCorbeel** [comment]
Your RAM and VRAM are the same?

For 24GB Vram you need something like 64gb ram

SO FOR 32GB Vram, which is a size I have not heard of so likely incorrect you would need 76GB Vram or something.

**Reply by ltdrdata** [reply]
In the case of Mac devices, they have an integrated memory structure where VRAM and RAM are unified.
Therefore, in the case of Macs, you can think of the entire memory as being composed solely of VRAM.

**Comment by ltdrdata** [comment]
Since the Sonoma upgrade, various issues have been occurring on Macs.
There are reports that downgrading to Torch 2.3.1 resolves these issues, so you might want to try that.

https://github.com/comfyanonymous/ComfyUI/issues/4165#issuecomment-2265861147
https://github.com/comfyanonymous/ComfyUI/issues/4165#issuecomment-2265880027

**Reply by movelikeriver** [reply]
thank you for your response! it was already running with 2.3.1
most of the time, it got error:
```
ComfyUI/nodes.py:1487: RuntimeWarning: invalid value encountered in cast
  img = Image.fromarray(np.clip(i, 0, 255).astype(np.uint8))
```

read through this thread #3521 , and tried the command below, modified ksampler, still didint work 
```
env TOKENIZERS_PARALLELISM=false python main.py --force-upcast-attention
```


the `TOKENIZERS_PARALLELISM` is due to
```
huggingface/tokenizers: The current process just got forked, after parallelism has already been used. Disabling parallelism to avoid deadlocks...
To disable this warning, you can either:
	- Avoid using `tokenizers` before the fork if possible
	- Explicitly set the environment variable TOKENIZERS_PARALLELISM=(true | false)                   
```
